gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -731,8 +731,6 @@ case class ColumnarOverrideRules(session: SparkSession)
   private lazy val transformPlanLogLevel = GlutenConfig.getConf.transformPlanLogLevel
   @transient private lazy val planChangeLogger = new PlanChangeLogger[SparkPlan]()
 
-  // Tracks whether the columnar rule is called through AQE.
-  private var isAdaptiveContext: Boolean = false
```

Comment:
Thanks so much for your fix!
Can we just use `ThreadLocal` type for `isAdaptiveContext` to fix this issue?

Reply:
`localProperties` is also a `ThreadLocal` parameter, using `localProperties` is OK to me.

Reply:
@PHILO-HE 
I think `LocalProperty `is more suitable here, and there will be some similar variables that need to be added in the future.